packages/web-components/src/utils/element-internals.ts

```diff
@@ -60,3 +64,46 @@ export function toggleState(elementInternals: ElementInternals, state: string, f
   // @ts-expect-error - Baseline 2024
   elementInternals.states.delete(state);
 }
+
+/**
+ * Check if a given attribute value is a valid state. Attribute values are often kebab-cased, so this function converts
+ * the kebab-cased `state` to camelCase and checks if it exists in as a key in the `States` object.
+ *
+ * @param States  - the object containing valid states for the attribute
+ * @param state - the state to check
+ * @returns true if the state is in the States object
+ * @internal
+ */
+export function hasMatchingState(States: Record<string, string> | undefined, state: string | undefined): boolean {
```

Comment:
Can we rename this to `isValidState()` to avoid any confusion with `CustomState.has()`? Also maybe `State` argument can be `ValidState`?

Reply:
`validState` and `userInvalidState` states both already exist as states, so I chose `hasMatchingState` to avoid confusion with those and other form validation-related things.